Makefile

```diff
@@ -30,3 +30,47 @@ shellcheck-gha:
 generate-gha-workflows:
 	./.github/scripts/generate_linux_ci_workflows.py
 	$(MAKE) shellcheck-gha
+
+setup_lint:
+	python tools/actions_local_runner.py --file .github/workflows/lint.yml \
+	 	--job 'flake8-py3' --step 'Install dependencies'
+	python tools/actions_local_runner.py --file .github/workflows/lint.yml \
+	 	--job 'cmakelint' --step 'Install dependencies'
+	pip install jinja2
+
+quick_checks:
+# TODO: This is broken when 'git config submodule.recurse' is 'true'
+	@python tools/actions_local_runner.py \
+		--file .github/workflows/lint.yml \
+		--job 'quick-checks' \
+		--step 'Ensure no trailing spaces' \
+		--step 'Ensure no tabs' \
+		--step 'Ensure no non-breaking spaces' \
+		--step 'Ensure canonical include' \
+		--step 'Ensure no unqualified noqa' \
+		--step 'Ensure no direct cub include' \
+		--step 'Ensure correct trailing newlines'
+
+flake8:
+	@python tools/actions_local_runner.py \
+		--file .github/workflows/lint.yml \
+		--job 'flake8-py3' \
+		--step 'Run flake8'
+
+mypy:
+	@python tools/actions_local_runner.py \
+		--file .github/workflows/lint.yml \
+		--job 'mypy' \
+		--step 'Run mypy'
+
+cmakelint:
+	@python tools/actions_local_runner.py \
+		--file .github/workflows/lint.yml \
+		--job 'cmakelint' \
+		--step 'Run cmakelint'
+
+clang_tidy:
+	echo "clang-tidy local lint is not yet implemented"
+	exit 1
+
+lint: flake8 mypy quick_checks cmakelint shellcheck-gha generate-gha-workflows
```

Comment:
Should this also include setup_lint?

Reply:
I think since it's mostly going to be regular PyTorch devs using this it doesn't need to do everything (i.e spend the time running those steps). `pip` also still spits out a bunch of output even when everything is already installed so it doesn't look great

We could have another wrapper script that checks which things are installed and only runs the actual `pip install` if it needs to, but I'll leave that for later

Reply:
Yeah I think leaving that to be a followup step is the best approach here